File: silverwind-frontend/src/app/core/services/auth.service.ts
import { Injectable, inject } from '@angular/core';
import { ApiService } from './api.service';
import { AuthStore } from '../stores/auth.store';
import { User, AuthResponse } from '../models/auth.model';
import { tap } from 'rxjs/operators';

@Injectable({
  providedIn: 'root',
})
export class AuthService {
  private api = inject(ApiService);
  private store = inject(AuthStore);

  login(credentials: { email: string; password: string }) {
    return this.api
      .post<AuthResponse>('/auth/login', credentials)
      .pipe(tap((response) => this.store.login(response)));
  }

  refreshUser() {
    return this.api.get<User>('/auth/me').pipe(
      tap((user) => {
        this.store.updateUser(user as User);
      }),
    );
  }

  registerVendor(data: any) {
    return this.api.post('/auth/register-vendor', data);
  }

  logout() {
    this.store.logout();
    // Optional: Call network endpoint if needed
  }
}

File: silverwind_flutter/lib/core/services/auth_service.dart
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../models/auth_models.dart';
import '../providers/auth_provider.dart';
import 'api_service.dart';

/// Auth service matching Angular's AuthService.
class AuthService {
  final ApiService _api;
  final AuthNotifier _authNotifier;

  AuthService(this._api, this._authNotifier);

  /// Login with email and password
  Future<AuthResponse> login({
    required String email,
    required String password,
  }) async {
    final response = await _api.post<Map<String, dynamic>>(
      '/auth/login',
      body: {'email': email, 'password': password},
    );
    final authResponse = AuthResponse.fromJson(response);
    _authNotifier.login(authResponse);
    return authResponse;
  }

  /// Refresh current user data
  Future<User> refreshUser() async {
    final userData = await _api.get<Map<String, dynamic>>('/auth/me');
    final user = User.fromJson(userData);
    _authNotifier.updateUser(user);
    return user;
  }

  /// Register vendor
  Future<dynamic> registerVendor(Map<String, dynamic> data) async {
    return await _api.post('/auth/register-vendor', body: data);
  }

  /// Logout
  void logout() {
    _authNotifier.logout();
  }
}

// ===== Provider =====
final authServiceProvider = Provider<AuthService>((ref) {
  final api = ref.watch(apiServiceProvider);
  final authNotifier = ref.watch(authProvider.notifier);
  return AuthService(api, authNotifier);
});
